src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -59,9 +60,11 @@ bool CAveragedPerceptron::train_machine(CFeatures* data)
 	for (int32_t i=0; i<num_feat; i++)
 		w[i]=1.0/num_feat;
 
+	CSignal::clear_cancel();
+
 	//loop till we either get everything classified right or reach max_iter
 
-	while (!converged && iter<max_iter)
+	while (!(CSignal::cancel_computations()) && (!converged && iter<max_iter))
 	{
```

Comment:
Yeah sure, but what should the message contain ? Which variables need to be highlighted..